### ISSUE TITLE
Testing codes with ConfigurationManager.AppSettings

### ISSUE THREAD

**Issue by zpbappi** [zpbappi/zpbappi.github.io#1]
Auto-generated issue to track comments for the post [Testing codes with ConfigurationManager.AppSettings](http://zpbappi.com/testing-codes-with-configurationmanager-appsettings/) in my [blog](http://zpbappi.com).

**Comment by rahulpnath** [comment]
Hey Bappi,

Nice one!

I would go one level further ([Refactoring to Improve Testability: Removing Unnecessary Dependencies](http://www.rahulpnath.com/blog/refactoring-to-improve-testability-removing-unnecessary-dependencies/)) and remove the dependency with the IAppSettings all together and just have the value that I am dependent on to be injected in. In this case the value of app.name. This reduces the dependency with the configuration name itself and also removes any need to mock in my tests. There can be a totally different class/factory to construct MyService class for me and would belong in the same project as the configuration files lives in.

**Comment by zpbappi** [comment]
Good idea. Didn't think about this level of abstraction back then. Thanks for pointing out. :+1:
